fabrictestbed_extensions/fablib/network_service.py

```diff
@@ -789,6 +789,25 @@ def get_available_ips(
             logging.warning(f"Failed to get_available_ips: {e}")
             return None
 
+    def get_public_ips(self) -> List[IPv4Address] or List[IPv6Address] or None:
+        """
+        Get list of public IPs assigned to the FabNetv*Ext service
+        :return: List of Public IPs
+        :rtype: List[IPv4Address] or List[IPv6Address] or None
+        """
+        if self.get_fim_network_service().labels is not None:
+            if self.get_fim_network_service().labels.ipv4 is not None:
+                result = []
+                for x in self.get_fim_network_service().labels.ipv4:
+                    result.append(IPv4Address(x))
+                return result
+            elif self.get_fim_network_service().labels.ipv6 is not None:
+                result = []
+                for x in self.get_fim_network_service().labels.ipv6:
+                    result.append(IPv6Address(x))
+                return result
```

Comment:
This seems to prefer IPv4 addresses -- can I ask for just IPv4 or IPv6 addresses?  Would that even be a good idea?

Reply:
Labels.ipv4 and Labels.ipv6 in the FIM object are never present together. Only one of them would be set depending on if the NS is of type FabNetV4Ext or FabNetV6Ext respectively.